library-best-practices/JLBP-7.md

```diff
@@ -0,0 +1,37 @@
+[JLBP-7] Make breaking transitions easy
+---------------------------------------
+
+When a library makes an in-place breaking change (meaning that the
```

Comment:
Begin by defining a breaking change. E.g. "a breaking change is a chaneg to the public or protected API of an existing class."
Would an out-of-place change ever be breaking?

Reply:
I think in a virtual sense, you can consider a transition from one major version to the next of an artifact that renames the Java package to be a breaking change to users who consider the new major version to be the true successor of the prior version, since they are forced to make code changes to adopt it. If you don't consider a successor with a renamed package to be the "same" library, then you have to consider the prior major version to be either an abandoned library or a deprecated library on long-term support.